File: aoj/volume100/csharp/A10032.cs
using System;
using System.Collections.Generic;

class AOJ
{
	static void Main()
	{
		var stack = new Stack<string>();
		while (true)
		{
			var instruction = Console.ReadLine().Split();
			var operation = instruction[0];
			if (operation == "quit")
				break;

			string block;
			switch (operation)
			{
				case "push":
					block = instruction[1];
					stack.Push(block);
					break;
				case "pop":
					block = stack.Pop();
					Console.WriteLine(block);
					break;
			}
		}
	}
}

File: aoj/volume100/scala/A10032.scala
object Main {
  var stack = scala.collection.mutable.Stack[String]()
  def main(args: Array[String]) {
    val instructions = Iterator.continually(scala.io.StdIn.readLine).takeWhile(_ != "quit")
    for (instruction <- instructions) {
      val input = instruction split " "
      val operation = input(0)
      if (operation == "push") {
        val block = input(1)
        stack.push(block)
      }
      if (operation == "pop") {
        val block = stack.pop()
        println(block)
      }
    }
  }
}
